config/scootconfig/modules.go

```diff
@@ -45,20 +45,48 @@ func (c *ClusterLocalConfig) Create() (*cluster.Cluster, error) {
 }
 
 type WorkersThriftConfig struct {
-	Type string
+	Type               string
+	PollingPeriod      string // will be parsed to a time.Duration
+	EnforceTaskTimeout bool
+	TaskTimeout        string // will be parsed to a time.Duration
 }
 
-const pollingPeriod = time.Duration(250) * time.Millisecond
+const defaultPollingPeriod = time.Duration(250) * time.Millisecond
+const defaultTaskTimeout = time.Duration(30) * time.Minute
+
+func (c *WorkersThriftConfig) Create(
+	tf thrift.TTransportFactory,
+	pf thrift.TProtocolFactory) (worker.WorkerFactory, error) {
+
+	pp := defaultPollingPeriod
+	tt := defaultTaskTimeout
+	var err error
+
+	// apply defaults
+	if c.PollingPeriod != "" {
+		pp, err = time.ParseDuration(c.PollingPeriod)
+		if err != nil {
+			return nil, err
+		}
+	}
 
-func MakeRpcWorkerFactory(tf thrift.TTransportFactory, pf thrift.TProtocolFactory) worker.WorkerFactory {
-	return func(node cluster.Node) worker.Worker {
-		c := client.NewClient(tf, pf, string(node.Id()))
-		return workers.NewPollingWorker(c, pollingPeriod)
+	if c.TaskTimeout != "" {
+		tt, err = time.ParseDuration(c.TaskTimeout)
+		if err != nil {
+			return nil, err
+		}
 	}
+
+	wf := func(node cluster.Node) worker.Worker {
```

Comment:
Oh, based on the name I was assuming this was more related to workerserver startup, not task run. If we did log levels this might be a debug or trace log, but nbd for now.